pynautobot/core/endpoint.py

```diff
@@ -690,3 +690,39 @@ def run(self, *args, api_version=None, **kwargs):
         ).post(args[0] if args else kwargs)
 
         return response_loader(req, self.return_obj, self)
+
+
+class GraphqlEndpoint(Endpoint):
+    """Extend Endpoint class to support run method for graphql queries."""
+
+    def run(self, *args, **kwargs):
+        """Runs a saved graphql query based on the query_id provided.
+
+        Takes a kwarg of `query_id` to specify the query that should be run.
+
+        Args:
+            *args (str, optional): Used as payload for POST method
+                to the API if provided.
+            **kwargs (str, optional): Any search argument the
+                endpoint accepts can be added as a keyword arg.
+
+        Returns:
+            An API response from the execution of the saved graphql query.
+
+        Examples:
+            To run a query with the following variables:
+            >>> query = nb.extras.graphql_queries.get("Example")
+            >>> query.run(
+                    variables={"foo": "bar"})
+                )
+        """
+
+        if not kwargs.get("query_id"):
+            raise ValueError('Keyword Argument "query_id" is required to run a query.')
+        query_run_url = f"{self.url}/{kwargs['query_id']}/run/"
+
+        return Request(
+            base=query_run_url,
+            token=self.token,
+            http_session=self.api.http_session,
+        ).post(args[0] if args else kwargs)
```

Comment:
What happens if you run a query without any required variables? Is the error message descriptive enough to pass through to the end user or should we catch the error and reword it?

Reply:
It's currently caught with the `query.RequestError` exception, but the API message is fairly descriptive:
```
  File "/home/ntc/pynautobot/connect.py", line 26, in <module>
    print(variable_query.run())
  File "/home/ntc/pynautobot/pynautobot/models/extras.py", line 50, in run
    return GraphqlEndpoint(self.api, self.api.extras, "graphql_queries").run(query_id=self.id, **kwargs)
  File "/home/ntc/pynautobot/pynautobot/core/endpoint.py", line 728, in run
    ).post(args[0] if args else kwargs)
  File "/home/ntc/pynautobot/pynautobot/core/query.py", line 398, in post
    return self._make_call(verb="post", data=data)
  File "/home/ntc/pynautobot/pynautobot/core/query.py", line 294, in _make_call
    raise RequestError(req)
pynautobot.core.query.RequestError: The request failed with code 400 Bad Request: {'errors': [{'message': 'Variable "$devicename" of required type "[String]!" was not provided.', 'locations': [{'line': 1, 'column': 15}]}]}
```

Reply:
Ok, that is fine. Thanks!